src/contracts/IdleRouter.sol

```diff
@@ -95,28 +97,32 @@ contract IdleRouter is OwnableUpgradeable {
     /**
      * @notice base function for depositing tokens into the CDO Tranches. Note: approvals
      * must be made before this call to allow the router to move your assets on your behalf
-     * @param _token the underlying token for the CDO
+     * @param _cdo address of the CDO contract
      * @param _amount of the underlying token to deposit
      * @param isAATranche set to true to deposit into an AA Tranche,
      * set to false to deposit into a BB Tranche 
      */
     function _deposit(
-        address _token,
+        address _cdo,
         uint256 _amount,
         bool isAATranche
     ) internal {
-        address idleCDOAddress = IIdleRegistry(idleRegistry).tokenToIdleCDO(
-            _token
+        address tokenAddress = IIdleRegistry(idleRegistry).idleCdoToToken(
+            _cdo
         );
-        require(idleCDOAddress != address(0), "IdleRouter: INVALID_TOKEN");
+
+        require(IIdleRegistry(idleRegistry).isValidCdo(_cdo), "IdleRegistry: INVALID_CDO");
```

Comment:
A few things here

1) Since we already have the address, it would be more gas efficient to just check that it's not the zero address versus making the additional external call.
2) The error message should start with IdleRegistry since that is where it is originating.

Reply:
sorry, meant IdleRouter (not registry)